haskala/modules/custom/haskala_terms/plugins/content_types/terms_list/terms_list.inc

```diff
@@ -0,0 +1,175 @@
+<?php
+
+$plugin = array(
+  'title' => t('Terms list'),
+  'description' => t('Display a terms list starting from some letter.'),
+  'category' => t('Haskala'),
+  'hook theme' => 'haskala_terms_terms_list_content_type_theme',
+  'required context' => new ctools_context_optional(t('Letter'), 'string'),
+  'defaults' => array(
+    'vocabulary' => 'Cities',
+  ),
+);
+
+/**
+ * Render callback.
+ */
+function haskala_terms_terms_list_content_type_render($subtype, $conf, $args, $context) {
+  $variables = array();
+
+  $argument = empty($context->data) ? 'A' : strtoupper($context->data);
+
+  $vocab_name = $conf['vocabulary'];
+
+  $vocabulary = taxonomy_vocabulary_machine_name_load($vocab_name);
+
+  $vid = $vocabulary->vid;
+
+  // Get english alphabet.
+  $letters_en = range('A', 'Z');
+
+  // Get hebrew alphabet;
+  $letters_he = haskala_book_hebrew_alphabet_without_sofit();
+
+  $count = 0;
+  $blocks = array();
+  $limit = 50;
+
+  // Get 50 terms of certain vocabulary, starting from current letter. If there
+  // are less then 50 items - take terms starting from next letter by alphabet.
+  do {
```

Comment:
This duplicates code from the books plugin. Please create a function that does this loop.